### PR TITLE
Publishing Wizard improvements #5497

### PR DESCRIPTION
**Pull request by edloidas**
Replaced a single warning line with more complex issues state bar, that will be universal among multiple dialogs.
Replaced issues state bar in publish dialog.
Made Content resource return number of not publishable content.
Made Content resource return number of not publishable content.
Most of the markChecking() calls are made for smoother UI transitions.

### CODE REVIEW

**Comment by alansemenov** [comment]
@vbradnitski please review the Java part

**Review by vbradnitski — APPROVED** [review]

**Comment by ashklianko** [comment]
When opening a publishing dialog with many items multiple items it may show for a short moment that all checks are passed and green:

https://user-images.githubusercontent.com/4255706/206406227-c5369af9-947a-48c5-97b0-9e5b34550bf9.mp4

**Comment by ashklianko** [comment]
Don't we need the same status bar in ContentUnpublishDIalog?

**Comment by ashklianko** [comment]
Looks good @edloidas !

File: modules/lib/src/main/resources/assets/js/app/dialog/DialogErrorsStateBar.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            className: 'resolved-entry',
+            text: resolvedText,
+            iconClass: 'icon-checkmark',
+        });
+
+        this.updateResolvedState();
+    }
+
+    protected updateResolvedState(): void {
+        const isResolved = !this.errorsEntries.some(entry => entry.isActive());
```
Comment (ashklianko) [inline modules/lib/src/main/resources/assets/js/app/dialog/DialogErrorsStateBar.ts]: `const isResolved: boolean = !this.errorsEntries.some((entry: DialogErrorStateEntry) => entry.isActive());`
I believe we have agreed long ago to always declare types on variables to make it easier to read, even though some types can be inferred, however in this case it requires to go and check what is entry's type

File: modules/lib/src/main/resources/assets/js/app/dialog/DialogErrorsStateBar.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        entry.onActiveStateChange(() => this.updateResolvedState());
+        this.updateResolvedState();
+
+        entry.onCheckingStateChange(() => this.updateChecking());
+
+        return entry;
+    }
+
+    setEnabled(enabled: boolean): void {
+        this.errorsEntries.forEach(entry => entry.setActionEnabled(enabled));
```
Comment (ashklianko) [inline modules/lib/src/main/resources/assets/js/app/dialog/DialogErrorsStateBar.ts]: Same here: `entry` is missing type

File: modules/lib/src/main/resources/assets/js/app/dialog/DialogErrorsStateBar.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        this.addClass(Modifiers.FAILED);
+    }
+
+    reset(): void {
+        this.errorsEntries.forEach(entry => entry.reset());
+        this.removeClass(Modifiers.FAILED);
+    }
+
+    markChecking(checking: boolean): void {
+        this.errorsEntries.forEach(entry => entry.markChecking(checking));
```
Comment (ashklianko) [inline modules/lib/src/main/resources/assets/js/app/dialog/DialogErrorsStateBar.ts]: Same here: `entry` is missing type

File: modules/lib/src/main/resources/assets/js/app/dialog/DialogErrorsStateBar.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    }
+
+    toggleHideIfResolved(hideIfResolved: boolean): void {
+        if (this.hideIfResolved !== hideIfResolved) {
+            this.hideIfResolved = hideIfResolved;
+            const isResolved = !this.errorsEntries.some(entry => entry.isActive());
```
Comment (ashklianko) [inline modules/lib/src/main/resources/assets/js/app/dialog/DialogErrorsStateBar.ts]: Types are missing

**Comment by alansemenov** [comment]
> Don't we need the same status bar in ContentUnpublishDIalog?

Correct, but this will be done in a separate task.

File: modules/lib/src/main/resources/assets/js/app/dialog/DialogErrorsStateBar.ts (hunk 1 above)
Comment (edloidas) [reply modules/lib/src/main/resources/assets/js/app/dialog/DialogErrorsStateBar.ts]: Oh boy, we haven't even got started. Guess we'll have to solve this with a one-on-one fight with knifes.

**Comment by edloidas** [comment]
Fixed the errors bar state problem.
It appears, that the initial problem behind the invalid errors bar was there for quite a long time and by using the debounced onLoadingFinished handlers just made it worse.
Because of the unholy mix of multiple debounced handlers, events and promises, that are used simultaneously, the handlers were triggered when they must not (visually this problem was affecting loading mask and controls lock state earlier).

Closing the dialog was triggering unnecessary delayed notifications, that had effect if you were quick enough to open it before the close callback were actually done running. Or you can just open the dialog, close it, and open again (in case the background checks will take long enough) to get double the notifications that will break the UI.

If we are to use it somewhere else in the future or add more functionality, I would recommend to write the publish processor from scratch if there'll be time.

**Comment by alansemenov** [comment]
Moved to https://github.com/enonic/app-contentstudio/pull/5562

**Pull request closed without merging**